src/pykiso/test_coordinator/test_suite.py

```diff
@@ -364,6 +371,69 @@ def __init__(
         # add sorted test case list to test suite
         self.addTests(test_case_list)
 
+        self.failed_suite_setups = set()
+
+    def check_suite_setup_failed(self, test: BasicTest, result: TestResult) -> None:
+        """Check if the suite setup has failed and store failed suite id.
+
+        :param test: test to check
+        :param result: unittest result object
+        """
+        if isinstance(test, BasicTestSuiteSetup):
```

Comment:
I do not understand the link between the `test` and `result`. How does the information that test is an instance of BasicTestSuiteSetup ensure that result does contain "failure" as attribute.

Reply:
See my other comment where i explain what information result stores.
The Unittest framework names the object result which is a little bit weird, because it does more than just storing a reseult.

For you to know it stores all the test results which has been run up to that point.

BasicTestSuiteSetup is a pykiso testcase which we use to perform the suite setup.